torch/utils/backend_registration.py

```diff
@@ -77,7 +80,9 @@ def rename_privateuse1_backend(backend_name: str) -> None:
         # to implement torch.ones.
         >>> a = torch.ones(2, device="foo")
         """
-    return _rename_privateuse1_backend(backend_name)
+    _rename_privateuse1_backend(backend_name)
+    global _privateuse1_backend_name
```

Comment:
Why do we need this to be a global now? It doesn't look like it was previously.

Reply:
> Why do we need this to be a global now? It doesn't look like it was previously.

yeah, there are two points: 1. we should use `torch._C._get_privateuse1_backend_name()` to get the backend-name, but it will cause an error with `torch.jit.script`.
2. In the func of `_get_custom_mod_func`, we need to get the device-module by `getattr(torch, backend-name)`, use the global variable will be faster than ``torch._C._get_privateuse1_backend_name()` which could be call many times. @bdhirsh